client/ayon_core/pipeline/staging_dir.py

```diff
@@ -0,0 +1,223 @@
+from collections import namedtuple
+
+from ayon_core.lib import Logger, filter_profiles, StringTemplate
+from ayon_core.settings import get_project_settings
+
+from .template_data import get_template_data
+from .anatomy import Anatomy
+from .tempdir import get_temp_dir
+
+
+StagingDir = namedtuple("StagingDir", ["dir", "persistent"])
```

Comment:
I would avoid using attribute `dir`, as it is built-in function, and would use dataclass only because of typehints.
```suggestion
@dataclass
class StagingDir:
    directory: str
    persistent: bool
```
Requires `from dataclasses import dataclass`